services/director-v2/src/simcore_service_director_v2/api/routes/computations.py

```diff
@@ -198,13 +199,16 @@ async def create_computation(  # noqa: C901, PLR0912
         min_computation_nodes: list[NodeID] = [
             NodeID(n) for n in minimal_computational_dag.nodes()
         ]
+        resource_usage_api = ResourceUsageApi.get_from_state(request.app)
```

Comment:
this is normally not the way FastAPI apps work. you should create a dependency see in the request itself, and FastAPI itself takes care of the dependencies. if you do not know how to do that ask @pcrespov as I will likely not be around this afternoon.

Also, if this is the resource_usage_client (as it seems to be), please call it like so. otherwise it sounds like it is some internal api

Reply:
@matusdrobuliak66 I think @sanderegg refers that we should use dependency injection to get api-clients and avoid exposing `request` in fastapi handlers. See as example the `rut_api` (which is the same type as yours`
https://github.com/ITISFoundation/osparc-simcore/blob/master/services/payments/src/simcore_service_payments/api/rest/_acknowledgements.py#L67-L76

Reply:
ok changed, thanks.